pkg/diagnose/connectivity/trace_request.go

```diff
@@ -24,10 +24,74 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/resolver"
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
+func init() {
+	diagnosis.Register("connectivity-datadog-core-endpoints", diagnose)
+}
+
+func diagnose(diagCfg diagnosis.DiagnoseConfig) []diagnosis.Diagnosis {
+
+	// Create domain resolvers
+	keysPerDomain, err := utils.GetMultipleEndpoints(config.Datadog)
+	if err != nil {
+		return []diagnosis.Diagnosis{
+			{
+				Result:      diagnosis.DiagnosisSuccess,
+				Name:        "Endpoints configuration",
+				Diagnosis:   "Misconfiguration of agent endpoints",
+				Remediation: "Please validate Agent configuration",
+				RawError:    err,
+			},
+		}
+	}
+
+	diagnoses := make([]diagnosis.Diagnosis, 0)
+
+	domainResolvers := resolver.NewSingleDomainResolvers(keysPerDomain)
+
+	client := forwarder.NewHTTPClient(config.Datadog)
+
+	// Send requests to all endpoints for all domains
+	for _, domainResolver := range domainResolvers {
+		// Go through all API Keys of a domain and send an HTTP request on each endpoint
+		for _, apiKey := range domainResolver.GetAPIKeys() {
+
+			for _, endpointInfo := range endpointsInfo {
+
+				domain, _ := domainResolver.Resolve(endpointInfo.Endpoint)
+
+				ctx := context.Background()
+				statusCode, responseBody, logURL, err := sendHTTPRequestToEndpoint(ctx, client, domain, endpointInfo, apiKey)
+
+				// Check if there is a response and if it's valid
+				report, reportErr := verifyEndpointResponse(statusCode, responseBody, err)
+				name := "Connectivity to " + logURL
+				if reportErr == nil {
+					diagnoses = append(diagnoses, diagnosis.Diagnosis{
+						Result:    diagnosis.DiagnosisSuccess,
+						Name:      name,
+						Diagnosis: fmt.Sprintf("Connectivity to `%s` is Ok\n%s", logURL, report),
+					})
+				} else {
+					diagnoses = append(diagnoses, diagnosis.Diagnosis{
+						Result:      diagnosis.DiagnosisFail,
+						Name:        name,
+						Diagnosis:   fmt.Sprintf("Connection to `%s` is falied\n%s", logURL, report),
+						Remediation: "Please validate Agent configuration and firewall to access " + logURL,
+						RawError:    err,
+					})
+				}
```

Comment:
```suggestion
				d := diagnosis.Diagnosis{
					Name: "Connectivity to " + logURL
				}
				if reportErr == nil {
					d.Result = diagnosis.DiagnosisSuccess
					d.Diagnosis = fmt.Sprintf("Connectivity to `%s` is Ok\n%s", logURL, report)
				} else {
					d.Result = diagnosis.DiagnosisFail
					d.Diagnosis = fmt.Sprintf("Connection to `%s` is falied\n%s", logURL, report)
					d.Remediation = "Please validate Agent configuration and firewall to access " + logURL
					d.RawError = err
				}
				diagnoses = append(diagnoses, d)
```
optional style change

Reply:
I like it. I did not particular like my construction. I am missing ternary expressions. Thank you.